src/main/java/tasklist/TaskList.java

```diff
@@ -0,0 +1,106 @@
+package tasklist;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import errors.InvalidIndexException;
+
+import localstorage.Storage;
+
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.Todo;
+
+public class TaskList {
+    private ArrayList<Task> tasks;
+    private Storage storage;
+
+    public TaskList(Storage storage) {
+        this.tasks = new ArrayList<Task>();
+        this.storage = storage;
+    }
+
+    public int size() {
+        return tasks.size();
+    }
+
+    public void loadFromStorgae() {
+        storage.loadTasks(tasks);
+    }
+
+    private void saveToStorage() throws IOException {
+        storage.saveTasks(tasks);
+    }
+
+    public List<String> listTasks() {
+        return tasks.stream()
+                .map(Task::getAsListItem)
+                .toList();
+    }
+
+    public Todo addTodo(String taskName) throws IOException{
```

Comment:
Nit: Whitespace before `{`

```suggestion
    public Todo addTodo(String taskName) throws IOException {
```

Reply:
Thanks for the spot! Will update accordingly.